google/cloud/internal/retry_loop_helpers.cc

```diff
@@ -106,6 +106,18 @@ Status RetryLoopCancelled(Status const& status, char const* location) {
   return Status(status.code(), std::move(message), std::move(ei));
 }
 
+StatusOr<std::chrono::milliseconds> BackoffOrBreak(Status const& status,
+                                                   char const* location,
+                                                   RetryPolicy& retry,
+                                                   BackoffPolicy& backoff,
+                                                   Idempotency idempotency) {
+  if (idempotency == Idempotency::kNonIdempotent) {
+    return RetryLoopNonIdempotentError(status, location);
+  }
+  if (retry.OnFailure(status)) return backoff.OnCompletion();
+  return RetryLoopError(status, location, retry.IsExhausted());
```

Comment:
I know this is just the "prepare" step, but I've been trying to make sense of the `RetryInfo` description ...
```
// Clients should wait until `retry_delay` amount of time has passed since
// receiving the error response before retrying.  If retrying requests also
// fail, clients should use an exponential backoff scheme to gradually increase
// the delay between retries based on `retry_delay`, until either a maximum 
// number of retries have been reached or a maximum retry delay cap has been
// reached.
```
In particular, I'm not sure what _"gradually increase the delay between retries based on `retry_delay`"_ means.  It doesn't look like the upcoming "heed RetryInfo in common loops" change does anything like that.

Reply:
> `RetryInfo` description [...] doesn't look like the upcoming "heed `RetryInfo` in common loops" change

Right. Bigtable has bastardized this old proto for their own purposes. I still argue for making this change to common (Option A in go/cloud-cxx:cbt-retry-info). To summarize:

- `RetryInfo` does not seem to be used much
- Even if it is used currently, we will only heed it by default in Bigtable
- Making Bigtable-specific changes in common is less offensive than what it would look like to make these changes in Bigtable
- Other services might adopt `RetryInfo` with Bigtable's interpretation